src/cargo/sources/git/utils.rs

```diff
@@ -1005,6 +1038,12 @@ pub fn fetch(
             // again. If it looks like any other kind of error, or if we've already
             // blown away the repository, then we want to return the error as-is.
             let mut repo_reinitialized = false;
+            // while shallow repos aren't officially supported, don't risk fetching them.
+            // We are in this situation only when `gitoxide` is cloning but then disabled to use `git2`
+            // for fetching.
+            if repo.is_shallow() {
```

Comment:
That's a great catch! 

When panicking in this branch, no test fails so that might mean it's not needed anymore. However, maybe the right test just isn't written yet. I am saying this because I had a feeling that a shallow crates index would still be in the same repository. Am I mistaken?

Depending on your answer I'd either try to write that missing test, or remove this portion of code.

Reply:
That's fair, as what I tried to say is: I don't know if a shallow clone of the crates index is at the same location like an unshallow one. If that was the case, older cargo's could encounter it. But if they did, then only *newer* cargos with this line would be able to deal with it, so I think the test that needs to be written is places shallow indices into their own directory.

Now that I write this, I am quite certain that I didn't implement such a thing as even now I am hesitant to break certain assumptions of crates like `crates-index` and probably even `crates-index-diff`. I guess that would, however, be preferable over letting older cargos encounter shallow repos.

*…time passes…*

Actually, there is a test for that and it is definitely writing the index into a directory suffixed `-shallow`, as the `ident()` code is shared among all git dependencies.

After all that it was clear that this condition was a useless leftover :) - thanks again, great catch!

Removed in a839431c86879bdaabf9b6df87e58dfedddaa371.